File: packintel_ai/frontend/src/types/simulator.ts
import { RecommendRequest, RankedMaterial, RecommendResponse } from './recommendation'

export interface SimulateRequest {
  base_query_id?: string
  food_commodity: string
  conditions: RecommendRequest
  demo_mode?: boolean
}

export interface DeltaResult {
  previous_top_material: string
  new_top_material: string
  score_change: number
  changed: boolean
  summary: string
}

export interface SimulateResponse extends RecommendResponse {
  simulation_id?: string
  results?: RankedMaterial[]
  delta_from_base?: DeltaResult
}

File: packintel_ai/backend/app/schemas/simulator.py
from pydantic import BaseModel
from typing import Optional, List
from app.schemas.recommendation import RecommendRequest, RankedMaterial


class SimulateRequest(BaseModel):
    base_query_id: Optional[str] = None
    food_commodity: str
    conditions: RecommendRequest
    demo_mode: Optional[bool] = None


class DeltaResult(BaseModel):
    previous_top_material: str
    new_top_material: str
    score_change: float
    changed: bool
    summary: str


class SimulateResponse(BaseModel):
    simulation_id: str
    results: List[RankedMaterial]
    delta_from_base: Optional[DeltaResult] = None
    processing_time_ms: int
